modules/javafx.web/src/main/native/Source/WebKitLegacy/java/WebCoreSupport/WebPage.cpp

```diff
@@ -2250,8 +2250,11 @@ JNIEXPORT jint JNICALL Java_com_sun_webkit_WebPage_twkProcessDrag
         case com_sun_webkit_WebPage_DND_DST_EXIT:
             dc.dragExited(*localMainFrame,WTFMove(dragData));
             return 0;
+        case com_sun_webkit_WebPage_DND_DST_ENTER:
+            return dragOperationToDragCursor(std::get<std::optional<WebCore::DragOperation>>(dc.dragEnteredOrUpdated(*localMainFrame, WTFMove(dragData))));
```

Comment:
This is identical to the body of the two cases below. It might be clearer to remove this return statement (line 2254) to avoid duplication.

Speaking of which, I presume that it is intentional to call `dragOperationToDragCursor` for all three of `DND_DST_ENTER`, `DND_DST_OVER`, and `DND_DST_CHANGE`?

Reply:
Yes, removed the duplicate call. In webkit-617.1, there were  calls to **dragEntered**  and **dragUpdated** for DND calls as follows:

      ` case com_sun_webkit_WebPage_DND_DST_ENTER:
            return dragOperationToDragCursor(dc.dragEntered(WTFMove(dragData)));`

       `case com_sun_webkit_WebPage_DND_DST_OVER:
        case com_sun_webkit_WebPage_DND_DST_CHANGE:
            return dragOperationToDragCursor(dc.dragUpdated(WTFMove(dragData)));`

In the current webkit-619.1 update, there has been refactoring and the apis **dragEntered**  and **dragUpdated** have been removed and instead **dragEnteredOrUpdated** has been used.  For our JAVA port adaptation, these changes have been missed and hence, the issue is observed. This fix addresses the issue.